src/Voron/Data/Tables/TableSchema.cs

```diff
@@ -328,31 +317,169 @@ public void Validate(FixedSizeSchemaIndexDef actual)
             }
         }
 
-        public TableSchema CompressValues(FixedSizeSchemaIndexDef etagSource, bool compress)
+        public class DynamicBTreeIndexDef : AbstractBTreeIndexDef
+        {
+            public override TableIndexType Type => TableIndexType.Dynamic;
+
+            public delegate ByteStringContext.Scope IndexValueAction(ByteStringContext context, ref TableValueReader value, out Slice slice);
```

Comment:
You were right, it's more about the key of indexed entries not about the values. So maybe `IndexEntryKeyGenerator` would be more suitable?

Reply:
done